pkg/controller/acmeorders/sync.go

```diff
@@ -505,24 +506,46 @@ func (c *controller) finalizeOrder(ctx context.Context, cl acmecl.Interface, o *
 		derBytes = block.Bytes
 	}
 
+	// Call to CreateOrderCert finalizes the ACME order. This call can only be made once.
 	certSlice, certURL, err := cl.CreateOrderCert(ctx, o.Status.FinalizeURL, derBytes, true)
-	// if an ACME error is returned and it's a 4xx error, mark this Order as
-	// failed and do not retry it until after applying the global backoff.
-	if acmeErr, ok := err.(*acmeapi.Error); ok {
-		if acmeErr.StatusCode >= 400 && acmeErr.StatusCode < 500 {
-			log.Error(err, "failed to finalize Order resource due to bad request, marking Order as failed")
-			c.setOrderState(&o.Status, string(cmacme.Errored))
-			o.Status.Reason = fmt.Sprintf("Failed to finalize Order: %v", err)
-			return nil
+
+	acmeErr, ok := err.(*acmeapi.Error)
+
+	// If finalizing the order returns a 4xx error then either we have
+	// already finalized this order or the order can be considered failed.
```

Comment:
I think we can restrict this to 403 as per the spec; we could also add a quote to the RFC 8555 page 48 so that we remember why 403.

Reply:
Thanks for pointing that out 👍🏼 

I was thinking about any possible edge cases where a non-403 4xx error could have been returned for an already finalized order, however could not think of any, so I have updated this bit of code to check if the order isn't already valid only if 403 error is returned